File: source/includes/insert-data.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson.Serialization.Conventions;
using MongoDB.Driver;

public class InsertData
{
  // Replace the following with your Atlas connection string      
  private const string MongoConnectionString = "<connection-string>";

  public static void Main(string[] args)
  {
    // Connect to your Atlas cluster
    var client = new MongoClient(MongoConnectionString);

    // Reference the database and collection to use
    var database = client.GetDatabase("gettingStarted");
    var peopleCollection = database.GetCollection<Person>("people");
    
    // Create a new document
    Person newPerson = new()
    {
        Name = new Name
        { 
          First = "Alan",
          Last = "Turing"
        },
        Birth = new DateTime(1912, 5, 23), // May 23, 1912                                                                                                                            
        Death = new DateTime(1954, 5, 7),  // May 7, 1954                                                                                                                                 
        Contribs = new string[] {"Turing machine", "Turing test", "Turingery"},
        Views = 1250000
    };
    
    // Insert the document into the specified collection
    peopleCollection.InsertOne(newPerson);

    // Find and return the document
    var filter = Builders<Person>.Filter
      .Eq(person => person.Name.Last, "Turing");

    var document = peopleCollection.Find(filter).FirstOrDefault();
    Console.WriteLine($"Document found:\n{document.ToBsonDocument()}");
  }
}

public class Person
{
    public ObjectId Id { get; set; }
    public Name Name { get; set; }
    public DateTime Birth { get; set; }
    public DateTime Death { get; set; }
    public string[] Contribs { get; set; }
    public int Views { get; set; }
}
public class Name
{
    public string First { get; set; }
    public string Last { get; set; }
}

File: source/includes/insert-data.java
import static com.mongodb.client.model.Filters.eq;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.result.InsertOneResult;
import com.mongodb.MongoException;

import java.util.Arrays;
import java.util.Date;
import java.time.Instant;
import org.bson.types.ObjectId;
import org.bson.Document;

public class InsertData {
    public static void main(String[] args) {
        // Replace the placeholder with your Atlas connection string
        String uri = "<connection-string>";

        // Connect to your Atlas Cluster and insert a document
        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // Reference the database and collection to use
            MongoDatabase database = mongoClient.getDatabase("gettingStarted");
            MongoCollection<Document> collection = database.getCollection("people");

            try {
                // Insert a new document into the specified collection
                InsertOneResult result = collection.insertOne(new Document()
                        .append("_id", new ObjectId())
                        .append("name", new Document().append("first", "Alan").append("last", "Turing"))
                        .append("birth", Date.from(Instant.parse("1912-05-23T00:00:00.000+00:00")))
                        .append("death", Date.from(Instant.parse("1954-05-07T00:00:00.000+00:00")))
                        .append("contribs", Arrays.asList("Turing machine", "Turing test", "Turingery"))
                        .append("views", 1250000));
            } catch (MongoException me) {
                System.err.println("Unable to insert due to an error: " + me);
            }
            // Find and return the document
            Document document = collection.find(eq("name.last", "Turing"))
                    .first();
            if (document == null) {
                System.out.println("No results found.");
            } else {
                System.out.println("Document found:");
                System.out.println(document.toJson());
            }
        }
    }
}
